nsls2forge_utils/all_feedstocks.py

```diff
@@ -0,0 +1,99 @@
+import requests
+import github3
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_all_feedstocks_from_github(organization):
+    '''
+    Gets all public feedstock repository names from the nsls-ii-forge organization.
+
+
+    Parameters
+    ----------
+    organization : str
+        Name of organization on GitHub.
+
+    Returns
+    -------
+    names : List
+        List of repository names that end with '-feedstock' (stripped).
+    '''
+    org = github3.organization(organization)
+    repos = org.repositories()
+    names = []
+    try:
+        for repo in repos:
+            name = repo.name
+            if name.endswith("-feedstock"):
+                name = name.split("-feedstock")[0]
+                logger.info(name)
+                names.append(name)
+    except github3.GitHubError:
+        msg = ["Github rate limited. "]
+        c = org.ratelimit_remaining()
+        if c == 0:
+            msg.append("API timeout.")
+        logger.warning(" ".join(msg))
+        raise
+    return names
+
+
+def get_all_feedstocks(organization, cached=False):
+    '''
+    Gets all feedstocks either from GitHub or from names.txt if flag is specified.
+
+    Parameters
+    ----------
+    organization : str
+        Name of organization on GitHub.
+    cached : bool
+        Specified if client wants to take repository names from names.txt.
+
+    Returns
+    -------
+    names : List
+        List of repository names that end with '-feedstock' (stripped).
+    '''
+    if cached:
+        logger.info("reading names")
+        with open("names.txt", "r") as f:
+            names = f.read().split()
+        return names
+
+    names = get_all_feedstocks_from_github(organization)
+    return names
+
+
+def main(args=None):
+    organization = 'nsls-ii-forge'
+    # see if json exists for active feedstocks
+    try:
+        logger.info("fetching active feedstocks from admin-migrations")
+        r = requests.get(
+            "https://raw.githubusercontent.com/nsls-ii-forge/admin-migrations/"
```

Comment:
Should we have the `admin-migrations` repo?

Reply:
After taking a second look, I don't think it will be necessary at the moment. Should I change it or leave it in for now? The functionality of it won't change since if the request fails it pulls the names from GitHub or names.txt.

Reply:
Let's remove it if it's not used.